modules/RTC/RTCUtils.js

```diff
@@ -1441,6 +1441,26 @@ class RTCUtils extends Listenable {
                         error => reject(error),
                         options);
                 } else if (hasDesktop) {
+                    if (options.screenSharingSourceDeviceId) {
+                        this.getUserMediaWithConstraints(
+                            [ 'video' ],
+                            stream => successCallback({
+                                desktop: {
+                                    sourceId: '',
+                                    screenType: 'camera',
+                                    stream
+                                }
+                            }),
+                            error => reject(error),
+                            {
+                                cameraDeviceId:
+                                    options.screenSharingSourceDeviceId
+                            }
+                        );
+
+                        return;
```

Comment:
Note that the above chunk is for the older gum flow. It does not have guards against exact device id being used because I was unsure of the ideal way to proceed.

Reply:
man this method sucks, i had to expand it like 3 or 4 times to figure out where `hasDesktop` came from.  i think i need a little help with what the context is when we enter this block (i.e. what things are true/not true)

Reply:
Sure. This flow is entered through JitsiMeetJS.createLocalTracks. Eventually it gets to here in RTCUtils.obtainAudioAndVideoPermissions to do the actual getUserMediaCall. obtainAudioAndVideoPermissions looks something like this:

```
if (firefox) {
    // special case handling
} else {
    const hasDesktop = if 'desktop' requested in user media array
    if (hasDesktop) {
        // remove 'desktop' from requested user media array
    }
    if (requested audio or video) {
        // do gum
        // some error handling
        // if hasDesktop try to get the desktop stream
    } else if hasDesktop {
        //  try to get the desktop stream
    }
}
```

Reply:
I didn't handle the request for desktop in the callback for audio/video. Maybe I should? I was hoping that bit of code is not exercised by jitsi-meet but maybe it's something I should cover anyway.

Reply:
thanks. is `options.screenSharingSourceDeviceId` something entirely new?

Reply:
Yeah, screenSharingSourceDeviceId is the new thing that signals to the lib to use a camera for the desktop sharing input instead of going through normal desktop sharing.